src/integration/resources/inprocess/coordinator.go

```diff
@@ -0,0 +1,278 @@
+		tmpDirs: tmpDirs,
+	}
+	coord.start()
+
+	return coord, nil
+}
+
+func (c *coordinator) start() {
+	interruptCh := make(chan error, 1)
+	shutdownCh := make(chan struct{}, 1)
+
+	go func() {
+		server.Run(server.RunOptions{
+			Config:      c.cfg,
+			InterruptCh: interruptCh,
+			ShutdownCh:  shutdownCh,
+		})
+	}()
+
+	c.interruptCh = interruptCh
+	c.shutdownCh = shutdownCh
+}
+
+func (c *coordinator) GetNamespace() (admin.NamespaceGetResponse, error) {
+	return c.client.GetNamespace()
+}
+
+func (c *coordinator) WaitForNamespace(name string) error {
+	return c.client.WaitForNamespace(name)
+}
+
+func (c *coordinator) AddNamespace(request admin.NamespaceAddRequest) (admin.NamespaceGetResponse, error) {
+	return c.client.AddNamespace(request)
+}
+
+func (c *coordinator) UpdateNamespace(request admin.NamespaceUpdateRequest) (admin.NamespaceGetResponse, error) {
+	return c.client.UpdateNamespace(request)
+}
+
+func (c *coordinator) DeleteNamespace(namespaceID string) error {
+	return c.client.DeleteNamespace(namespaceID)
+}
+
+func (c *coordinator) CreateDatabase(request admin.DatabaseCreateRequest) (admin.DatabaseCreateResponse, error) {
+	return c.client.CreateDatabase(request)
+}
+
+func (c *coordinator) GetPlacement() (admin.PlacementGetResponse, error) {
+	return c.client.GetPlacement()
+}
+
+func (c *coordinator) WaitForInstances(ids []string) error {
+	return c.client.WaitForInstances(ids)
+}
+
+func (c *coordinator) WaitForShardsReady() error {
+	return c.client.WaitForShardsReady()
+}
+
+func (c *coordinator) Close() error {
+	defer func() {
+		for _, dir := range c.tmpDirs {
+			if err := os.RemoveAll(dir); err != nil {
+				c.logger.Error("error removing temp directory", zap.String("dir", dir), zap.Error(err))
+			}
+		}
+	}()
+
+	// TODO: confirm this works correctly when using an embedded coordinator
+	select {
+	case c.interruptCh <- errors.New("in-process coordinator being shut down"):
+	case <-time.After(5 * time.Second):
+		return errors.New("timeout sending interrupt. closing without graceful shutdown")
+	}
+
+	select {
+	case <-c.shutdownCh:
+	case <-time.After(1 * time.Minute):
+		return errors.New("timeout waiting for shutdown notification. server closing may" +
+			" not be completely graceful")
+	}
+
+	return nil
+}
+
+func (c *coordinator) ApplyKVUpdate(update string) error {
+	return c.client.ApplyKVUpdate(update)
+}
+
+func (c *coordinator) WriteCarbon(port int, metric string, v float64, t time.Time) error {
+	return c.client.WriteCarbon(fmt.Sprintf("http://0.0.0.0/%d", port), metric, v, t)
+}
+
+func (c *coordinator) WriteProm(name string, tags map[string]string, samples []prompb.Sample) error {
+	return c.client.WriteProm(name, tags, samples)
+}
+
+func (c *coordinator) RunQuery(
+	verifier resources.ResponseVerifier,
+	query string,
+	headers map[string][]string,
+) error {
+	return c.client.RunQuery(verifier, query, headers)
+}
+
+func updateCoordinatorPorts(cfg config.Configuration) (config.Configuration, error) {
```

Comment:
Combine updateCoordinatorPorts and updateCoordinatorFilepaths? they don't see to pull a lot of weight on their own.

Reply:
Any value in this receiving a ptr object and just modifying that directly?

Reply:
I kind of like the separation of concerns. I also expect both functions will grow as I haven't done a full audit of the coord config for each port and filepath that may need to be updated (just the ones to get started).

Re: ptr vs object -- all things being equal, we generally prefer passing a value rather than a pointer as outlined in this [article](https://goinbigdata.com/golang-pass-by-pointer-vs-pass-by-value/): 

> Passing by value often is cheaper
>
> Even though Go looks a bit like C, its compiler works differently. And C analogy does not always work with Go. 
> Passing by value in Go may be significantly cheaper than passing by pointer. This happens because Go uses escape > analysis to determine if variable can be safely allocated on function’s stack frame, which could be much cheaper 
> then allocating variable on the heap. Passing by value simplifies escape analysis in Go and gives variable a better 
> chance to be allocated on the stack.

Realistically, either would be fine here. Personal prefence likes the immutability of the call as it mirrors other go language constructs (e.g. append)